### PR TITLE
Add revert guidelines to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by willbarrett**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [X] Documentation

### SUMMARY

Adds a section to CONTRIBUTING.md that describes the responsibilities of engineers attempting to perform a revert on Superset.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

n/a

### TEST PLAN

n/a

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@mistercrunch @john-bodley @graceguo-supercat  @betodealmeida @etr2460 @craig-rueda @dpgaspar

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -226,6 +227,21 @@ If the PR passes CI tests and does not have any `need:` labels, it is ready for
 
 If an issue/PR has been inactive for >=30 days, it will be closed. If it does not have any status label, add `inactive`.
 
+## Revert Guidelines
+
+Reverting changes that are causing issues in the master branch is a normal and expected part of the development process. In an open source community, the ramifications of a change cannot always be fully understood. With that in mind, here are some considerations to keep in mind when considering a revert:
```
Comment (etr2460) [inline CONTRIBUTING.md]: Do you feel like age of the PR is something to consider? I'd say recent PRs are more acceptable to revert than very old ones for a couple reasons: it means the regression was discovered quickly so it likely affecting multiple users and it's unlikely the revert will cause issues due to other logic committed after the bad PR.

**Review by craig-rueda — APPROVED** [review]
LGTM

File: CONTRIBUTING.md (hunk 1 above)
Comment (willbarrett) [reply CONTRIBUTING.md]: Good point. Will add.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -226,6 +227,22 @@ If the PR passes CI tests and does not have any `need:` labels, it is ready for
...
+
+- **Availability of the PR author:** If the original PR author or the engineer who merged the code is highly available and can provide a fix in a reasonable timeframe, this would counter-indicate reverting.
+- **Severity of the Issue:** How severe is the problem on master? Is it keeping the project from moving forward? Is there user impact? What percentage of users will experience a problem?
+- **Size of the change being reverted:** Reverting a single small PR is a much lower-risk proposition than reverting a massive, multi-PR change.
+- **Age of the change being reverted:** Reverting a recently-merged PR will be more acceptable than reverting an older PR. A bug discovered in an older PR is unlikely to be causing widespread serious issues.
+- **Risk inherent in Reverting:** Will the reversion break critical functionality? Is the medicine more dangerous than the disease?
```
Comment (etr2460) [inline CONTRIBUTING.md]: sp nit: uncapitalize `Reverting`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -226,6 +227,22 @@ If the PR passes CI tests and does not have any `need:` labels, it is ready for
 
 If an issue/PR has been inactive for >=30 days, it will be closed. If it does not have any status label, add `inactive`.
 
+## Revert Guidelines
+
+Reverting changes that are causing issues in the master branch is a normal and expected part of the development process. In an open source community, the ramifications of a change cannot always be fully understood. With that in mind, here are some considerations to keep in mind when considering a revert:
+
+- **Availability of the PR author:** If the original PR author or the engineer who merged the code is highly available and can provide a fix in a reasonable timeframe, this would counter-indicate reverting.
+- **Severity of the Issue:** How severe is the problem on master? Is it keeping the project from moving forward? Is there user impact? What percentage of users will experience a problem?
```
Comment (etr2460) [inline CONTRIBUTING.md]: sp nit: uncapitalize `Issue`

**Review by etr2460 — APPROVED** [review]
lgtm with a couple spelling nits

**Review by dpgaspar — APPROVED** [review]

**Comment by graceguo-supercat** [comment]
Hi @willbarrett Thanks for adding this guideline. 
Do you have some `hard rules` to apply PR reverting? For example, break existed chart/queries, reported by multiple users, can't fix it for a pretty long time, .etc.

**Comment by willbarrett** [comment]
@graceguo-supercat we have intentionally avoided adding any hard and fast rules. This will always be a subjective decision. We have added responsibilities for the engineer reverting code changes around communication. I think this is the fundamental point where we've had disagreement in the community recently. Do you think that's sufficient or are there some hard-and-fast rules that you think we should add?

**Comment by mistercrunch** [comment]
I agree that this is multidimensional and fuzzy. It's hard to come up with universal rules here.

**Review by graceguo-supercat — APPROVED** [review]

**Pull request merged**